### DISCUSSION TITLE
can not call ui.page().__call(). no effect

### DISCUSSION THREAD

**Discussion by xxhank** [zauberzeug/nicegui#2024] [Q&A]
### Description

i want to create page automaticly as below, but it cannot work.

```py
for router_name in router_names:
    def index(router_name):  # client: Client
        with menu.frame(router_name):
            create_tabs(router_name, *page_tabs)
    ui.page(router_name).__call__(lambda: index(router_name))
```

**Comment by falkoschindler** [comment]
I'm not sure what exactly does not work, because your code isn't executable without `router_names` and `page_tabs` etc.
But in principle this should work:
```py
for route in ['/a', '/b', '/c']:
    def page_function(route=route):
        ui.label(f'Hello {route}!')
    ui.page(route)(page_function)
```
- `route=route` avoids the [famous](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-have-all-my-elements-the-same-value) late-binding issue.
- You don't need to call `__call__`. You can simply call the `ui.page(router_name)` with the `page_function`.
